Mouse support for old firmware

Abz79 opened the pull request:
Subject: Add Mouse Support for Old Firmware to Menu and Pointer Handling

Description:

This pull request introduces functionality to enable mouse interaction within the menu system.

Key changes include:

    Integration of mouse event handling logic.
    Modification of menu drawing functions to support mouse-specific timing or event processing.
    Mouse pointer now stays hidden if a timer is set until mouse movement is detected.
    Addition of controls for pointer display.

These changes allow users to navigate and select menu options using a mouse device on old firmwares, enhancing the user experience.

Commit:
Mouse support for old firmware

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RefindPlusRepo/RefindPlus?pullRequest=219) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/RefindPlusRepo/RefindPlus?pullRequest=219) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: Abz79<br/>:x: Abdallah Terro<br/><hr/>**Abdallah Terro** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RefindPlusRepo/RefindPlus?pullRequest=219) it.</sub>

Commit:
Merge branch 'GOPFix' into MouseFix

dakanji commented:
So there are a number of issues here:

What is expected is confirmation the mouse does not work with the current RefindPlus code and clear code that fixes this.
That is, it should be clearly apparent that changing X to Y or adding Z makes the mouse work where it did not before.
I cannot put my finger on this in the mass of changes and did not look at everything.

There should be none of:
- Integration of mouse event handling logic.
- Modification of menu drawing functions to support mouse-specific timing or event processing.
- Mouse pointer staying hidden if a timer is set until mouse movement is detected.
- Addition of controls for pointer display.

There is too much going on and too many changes proposed. Before spending any more time on this, please note that the mouse item is a very low priority item and given that it generally currently works on other units, any changes would need to be clear and simple.

As suggested before, you should consider proposing your changes to the rEFInd project instead and if it is adopted there, it will ultimately most likely come into RefindPlus. A simple clear modification as outlined at the beginning could be considered here.

You should also make sure that you can build the resulting code as I think it might be the case that your proposal may not build as it stands ... just a feeling and not tested.

All in all and while I appreciate the level of effort put in, I am afraid I must decline this as I cannot look into all the proposed changes.

-----

# Some General Notes

When preparing a pull request for any project, you should rebase your changes on top of the current code and not merge the current code on your changed code as done here. The merge means certain items from the project code may be missing from the final result. For instance, there was a fix by an earlier Pull Request and additional items I added later to improve this that have now been zapped.  

For future reference on the CLA item and other similar items that might flag such on this and other projects, you need to add the other email address to your GitHub profile. The CLA item here includes a link to read on this.

Abz79 commented:
I understand your concerns.
The thing is while debugging RedindPlus was not detecting any problem with the mouse and was waiting for the input. 
This usually works fine on new firmwares but on old firmwares it was receiving EFI_NOT_READY hence no mouse motion was drawn. 
But when checking about tab or running shell tool and going back to the loader the mouse would appear in a new position and stay there. So it was only updating during redraw. 
This fix will not only work with old firmwares but it will also not affect the new firmwares. So everything works fine.

I already rebased and updated all new GOPFix changes and rebuilt without any issues.

It's totally up to you to figure out how to make use of this option or if you decide not to use it at all. No problem whatsoever. I really enjoyed working on this fix and learned a lot in the process. I also checked the power of AI since I used GPT R1 and Gemini while testing and debugging and figured that Gemini is by miles the best :)

Thanks again boss and have a nice day

dakanji commented:
> I already rebased and updated all new GOPFix changes and rebuilt without any issues.

Rebasing is a specific Git item. What you did was a `merge commit` at the end.

I suppose it did build indeed because all the items added by the referenced earlier PR and the subsequent changes were deleted.
I thought it might fail because I had noticed all the associated headers had been deleted but yes, it would build since all the code that needed them had also been deleted.

Anyway, glad it was a worthwhile exercise in the end.
As mentioned, you could try upstream.
